### PR TITLE
Better Predictive Facing, Ping Meter, Ping Simulation

### LINKED ISSUE #2172: Need a way to simulate ping
# Description
in Unet HLAPI there was a way to turn on network simulator that was quite useful. Need a way to do it with Mirror.
This functionality is crucial for stress testing movement, push/pull and other predictive things

### LINKED ISSUE #2268: Ping meter is gone
# Description
image
it was in the bottom right corner and now it no longer is

### LINKED ISSUE #2432: Facing direction hicups on high ping clients B: 45
# Description

The gif below was recorded on a connection with a ping of 380ms.
The blue squares show when the facing direction gets updated with a wrong value

facingdirbug

Contributor will receive a bounty of 45usd on a successful merge of a solution that resolves the issue above.

### PR DESCRIPTION
**Pull request by chairbender**
### Purpose
Fixes #2432
Fixes #2268 
Fixes #2172 

Also fixes circuit board screwing in messages and issue where tile change manager was overwriting the change list on the server side when syncing a new player.

Simulation of lag is described here: https://github.com/unitystation/unitystation/wiki/Simulating-Lag

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- [x]  Any new / changed components follow the [Component Development Checklist](https://github.com/unitystation/unitystation/wiki/Component-Development-Checklist)
- [x]  Any new objects / items follow the [Creating Items and Objects Guide](https://github.com/unitystation/unitystation/wiki/Creating-Items-and-Objects%3A-Now-With-Prefab-Variants) (especially concerning the use of prefab variants)
- [x]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- [x]  This PR has been tested with round restarts.
- [ ]  This PR has been tested on moving / rotating / rotated-before-joining matrices (if applicable)

### Notes:
_Please enter any other relevant information here_

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Directional.cs [hunk 1]
```diff
@@ -249,23 +213,52 @@ public void TargetForceSyncDirection(NetworkConnection target)
...
 
     //syncvar hook invoked when server sends a client the new direction for this object
-    private void ServerDirectionChangeHook(Orientation dir)
+    private void SyncServerDirection(Orientation dir)
     {
 	    serverDirection = dir;
-	    if (!IgnoreServerUpdates)
+	    //we only change our direction if we're not local player (local player is always predictive)
+	    //and not explicitly ignoring server updates.
+	    if (!IgnoreServerUpdates && !IsLocalPlayer)
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Objects/Directional.cs:261]: NOTE: This was the only substantial change to Directional - adding !IsLocalPlayer

File: UnityProject/Assets/Scripts/UI/Lobby/GUI_LobbyDialogue.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace Lobby
 {
 	public class GUI_LobbyDialogue : MonoBehaviour
 	{
-		private const string DefaultServerAddress = "localhost";
+		private const string DefaultServerAddress = "127.0.0.1";
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/UI/Lobby/GUI_LobbyDialogue.cs:13]: NOTE: Needs to use loopback address or some ping simulation tools won't work

**Comment by chairbender** [comment]
Also @DooblyNoobly , I'd rather the bounty for the facing direction issue go back into the bounty pool if that's an option. I'd rather just put it back into the project instead of it being eaten up by transaction costs.

**Comment by DooblyNoobly** [comment]
roger that

File: UnityProject/Assets/Scripts/UI/Lobby/GUI_LobbyDialogue.cs (hunk 2 above)
Comment (fomalsd) [reply UnityProject/Assets/Scripts/UI/Lobby/GUI_LobbyDialogue.cs:13]: so `localhost` isn't recognized by some tools as a loopback address?

Comment (chairbender) [reply UnityProject/Assets/Scripts/UI/Lobby/GUI_LobbyDialogue.cs:13]: yes, as I documented in the linked Wiki page

Comment (chairbender) [reply UnityProject/Assets/Scripts/UI/Lobby/GUI_LobbyDialogue.cs:13]: er, maybe more accurate to just say it doesn't work with localhost, not sure whether that has to do with it being "recognized as a loopback address", more just that the tool doesn't seem to want to mess with things going over localhost but does as soon as you switch to 127.0.0.1

**Review by fomalsd — APPROVED** [review]
that's awesome, works really well!

**Comment by fomalsd** [comment]
oops, conflicts

**Pull request merged**